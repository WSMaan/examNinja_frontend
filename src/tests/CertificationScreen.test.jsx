import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import axios from 'axios';
import { act } from 'react';
import CertificationScreen from '../components/Quest.jsx'; // Adjust the import based on your file structure

// Mock the axios module
jest.mock('axios');

describe('CertificationScreen', () => {
    const mockTests = [
        { testId: 1, testName: 'Click here to start the Test Name - 50 Questions', numberOfQuestions: 50 },
        { testId: 2, testName: 'Another Test - 60 Questions', numberOfQuestions: 60 }
    ];

    // Mock sessionStorage to provide a token before all tests
    beforeAll(() => {
        sessionStorage.setItem('token', 'mockedToken');
    });

    beforeEach(() => {
        axios.get.mockResolvedValue({ data: mockTests });
    });

    test('renders the loading state initially', () => {
        render(
            <Router>
                <CertificationScreen />
            </Router>
        );
        expect(screen.getByText(/loading tests.../i)).toBeInTheDocument();
    });

    test('displays fetched tests after loading', async () => {
        await act(async () => {
            render(
                <Router>
                    <CertificationScreen />
                </Router>
            );
        });

        await waitFor(() => {
            expect(screen.getByText(/click here to start the test name - 50 questions/i)).toBeInTheDocument();
            expect(screen.getByText(/another test - 60 questions/i)).toBeInTheDocument();
        });
    });

    test('handles error state', async () => {
        axios.get.mockRejectedValue(new Error('Failed to load tests.'));

        await act(async () => {
            render(
                <Router>
                    <CertificationScreen />
                </Router>
            );
        });

        await waitFor(() => {
            expect(screen.getByText(/failed to load tests/i)).toBeInTheDocument();
        });
    });

    test('matches snapshot', () => {
        const { asFragment } = render(
            <Router>
                <CertificationScreen />
            </Router>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
