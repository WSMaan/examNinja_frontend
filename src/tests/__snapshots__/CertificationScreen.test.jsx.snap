// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CertificationScreen matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="tab-container"
  >
    <div
      class="MuiBox-root css-0"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 tab-title css-xhm7gi-MuiTypography-root"
      >
        ExamNinja Test Studio
      </p>
      <div
        class="MuiTabs-root css-ijz6m-MuiTabs-root"
      >
        <div
          class="MuiTabs-scrollableX MuiTabs-hideScrollbar css-oqr85h"
          style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
        />
        <div
          class="MuiTabs-scroller MuiTabs-hideScrollbar MuiTabs-scrollableX css-1hiyfpk-MuiTabs-scroller"
          style="margin-bottom: 0px;"
        >
          <div
            aria-label="Navigation Tabs"
            class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected css-1pe4ln8-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="0"
              type="button"
            >
              Home
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit css-1pe4ln8-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Certifications
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit css-1pe4ln8-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Test
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit css-1pe4ln8-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Review
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit css-1pe4ln8-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Progress
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit css-1pe4ln8-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Status
            </button>
          </div>
          <span
            class="MuiTabs-indicator css-1sj2jcg-MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
    </div>
    <div
      class="certification-box MuiBox-root css-0"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 instructions css-8zgow4-MuiTypography-root"
      >
        You have access to the following Question Banks and Tests. Please click on the test that you want to attempt.
      </h6>
      <div
        class="exam-header MuiBox-root css-0"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-8zgow4-MuiTypography-root"
        >
          EXPIRED - JA+ I V8 For OCA-JP-I SE8 (1Z0-808)
        </h6>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 loading css-1edfpdg-MuiTypography-root"
      >
        Loading tests...
      </p>
    </div>
  </div>
</DocumentFragment>
`;
